config/data_providers.yml

```diff
@@ -0,0 +1,10 @@
+shared:
+  exchange_rates:
+    provider: <%= ENV.fetch "EXCHANGE_RATES_PROVIDER", "synth" %>
+    key: <%= ENV.fetch "EXCHANGE_RATES_API_KEY", ENV["SYNTH_API_KEY"] %>
+  merchant_data:
+    provider: <%= ENV.fetch "MERCHANT_DATA_PROVIDER", "synth" %>
+    key: <%= ENV.fetch "MERCHANT_DATA_API_KEY", ENV["SYNTH_API_KEY"] %>
+  reale_state_valuations:
+    provider: <%= ENV.fetch "REAL_ESTATE_VALUATIONS_PROVIDER", "null" %>
+    key: <%= ENV["REAL_ESTATE_VALUATIONS_API_KEY"] %>
```

Comment:
I think I'd recommend something like this, instead:

```
providers:
  exhange_rates: <%= ENV.fetch "EXCHANGE_RATES_PROVIDER", "synth" %>
synth:
  key: ENV["SYNTH_API_KEY"]
```

Mostly, this lets each provider define it's own instantiation, but still lets your user define which provider they want to use. 

For example, I think it's Google Cloud whose library wants a JSON file of credentials, instead of an API key.

Like - hmm. Your abstraction _isn't_ "all providers are the same", it's, "all providers can produce the same info". So you don't need the configuration of each provider to be identical.

Reply:
We don't need the configuration of each provider to be identical. But it's rather nice that it is.

I can see some value in adding the provider names as keys here, instead of generic "CONCEPT_PROVIDER_API_KEY". Kind of annoying that the example file would get bigger with each provider.

I originally had even removed SYNTH_API_KEY, but left it in as a fallback because CI would break otherwise. I think if we want to remove the synth one we'd have to add the others as secrets to GH cc @zachgoll up to you!

Reply:
I don't think we need to solve for this right now. Any non-standard behavior can be addressed in the concrete provider initializer. Also not sure data providers would require much more than an API key usually — I don't anticipate Google Cloud or similar being a data provider in this capacity.

Might push up a commit where the provider classes know about their provider-specific env vars. Would certainly be nice to be able to instantiate them without further config. Will revisit after work.

Reply:
https://github.com/maybe-finance/maybe/pull/561/commits/ef7ced2e4c4fb752425e219a717ac90ae4424c1b